modules/codepipeline/roles.tf

```diff
@@ -146,10 +146,12 @@ resource "aws_iam_role_policy" "lambda_codebuild_role_policy" {
       ],
       "Resource": [
         "${aws_s3_bucket.codepipeline_bucket.arn}",
-        "${aws_s3_bucket.codepipeline_bucket.arn}/*"
+        "${aws_s3_bucket.codepipeline_bucket.arn}/*",
+        "${data.aws_s3_bucket.state_bucket.arn}",
```

Comment:
The codepipeline role needed access to the S3 bucket which contains the terraform state. So the arn is included here.